libherokubuildpack/examples/print_style_guide.rs

```diff
@@ -0,0 +1,224 @@
+use ascii_table::AsciiTable;
+use fun_run::CommandWithName;
+use indoc::formatdoc;
+use libherokubuildpack::output::style::{self, DEBUG_INFO, HELP};
+#[allow(clippy::wildcard_imports)]
+use libherokubuildpack::output::{
+    build_log::BuildLog,
+    section_log::{log_step, log_step_stream, log_step_timed},
+};
+use std::io::stdout;
+use std::process::Command;
+
+// Avoid cargo-clippy warnings: "external crate X unused in `print_style_guide`"
+use const_format as _;
+use crossbeam_utils as _;
+use flate2 as _;
+use lazy_static as _;
+use libcnb as _;
+use libcnb_test as _;
+use pathdiff as _;
+use pretty_assertions as _;
+use regex as _;
+use sha2 as _;
+use tar as _;
+use tempfile as _;
+use termcolor as _;
+use thiserror as _;
+use toml as _;
+use ureq as _;
+
+#[allow(clippy::too_many_lines)]
+fn main() {
+    println!(
+        "{}",
+        formatdoc! {"
+
+            Living build output style guide
```

Comment:
> Or at the very least, I think it would simplify/speed up this PR if this addition were kept separate, so we can have the discussion about it another time :-)

Done in manuel's PR, which I've pulled into this PR. 

This is my attempt at having non-stale guides. I'll save the reasons for that to a later discussion.